src/gui/properties/speedplotview.cpp

```diff
@@ -205,69 +214,65 @@ void SpeedPlotView::setGraphEnable(GraphID id, bool enable)
     viewport()->update();
 }
 
-void SpeedPlotView::pushPoint(const SpeedPlotView::PointData &point)
+void SpeedPlotView::pushPoint(const SpeedPlotView::SampleData &point)
 {
-    m_data5Min.push_back(point);
-    m_averager30Min.push(point);
-    m_averager6Hour.push(point);
-    m_averager12Hour.push(point);
-    m_averager24Hour.push(point);
+    for (Averager *averager : {&m_averager5Min, &m_averager30Min
```

Comment:
QHash will add overhead and QHash will be used in two places, and for some graph, there is the same average so we have to add conditions for that to save for double push

Reply:
It would be nice if adding another Averager only required changes in one place, rather than four, as it is now. But it looks like it's a little more complicated than I thought. So I won't insist on it. In addition, adding/removing an Averager is very rare.